### DISCUSSION TITLE
how to make a object (mass=0) move?

### DISCUSSION THREAD

**Discussion by zhang-qiang-github** [InteractiveComputerGraphics/PositionBasedDynamics#178] [Q&A]
I am learning examples/beam_model.py:

![image](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/assets/29271051/bdfd8ff5-b9b7-49de-ac8d-ecd9ce251749)

For this beam, the left is fixed by setting mass=0. And the other point is moving. 

I want the whole beam is moving along x axis. For example: 

![image](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/assets/29271051/0d6c4257-4c0f-46ed-9dcb-f48c9c6716d4)

The black indicate the beam at t, and the gray indicate the beam at t+1. I hope there are two moving for beam:
1. the fix point is moving along x axis. If the fix point is (0,0,0) at t, I hope it turn to be (vx, 0, 0) at t+1.
2. the beam continue bending because of the gravity.

How can I make it moving along x axis? Any suggestion is helpful.

**Comment by janbender** [comment]
You could animate the fixed points, e.g. by using a Python script:
https://positionbaseddynamics.readthedocs.io/en/latest/py_getting_started.html

The project also contains some examples that may help you.

**Reply by zhang-qiang-github** [reply]
Thank you very much for reply. But, “pyPBD/examples” provide very few examples. I can not find a similar demo with my situation.
